### DISCUSSION TITLE
Ploltly-resampler unable to load lttbc

### DISCUSSION THREAD

**Discussion by MIKKELLORENZ** [predict-idlab/plotly-resampler#135] [Q&A]
Dear predict-idlab,

I have been successful in getting plotly resampler to work, but it is very slow. Then I noticed in my cmd of my application the following:

_C:\ProgramData\Anaconda3\lib\site-packages\plotly_resampler\aggregation\aggregators.py:24: UserWarning: Could not import lttbc; will use a (slower) python alternative.
  warnings.warn("Could not import lttbc; will use a (slower) python alternative.")_

When I do **conda list** I have the follwing versions:

**plotly-resampler     0.8.1     pypi_0                        pypi**
**lttbc                        0.2.1     (py39h5d4886f_1)     conda-forage**

Any ideas what might cause this to occour? 

Best Mikkel

**Comment by jonasvdd** [comment]
Hi @MIKKELLORENZ,

Quoting my answer from the [plotly-community forum](https://community.plotly.com/t/plotly-resampler-visualize-large-time-series-using-plotly-dash/59097/12?u=jonvdrdo)
> This most likely means that our more efficient C-implementation of the LTTB data aggregation algorithm (i.e., lttbc) was not compiled successfully during installation of plotly-resampler.
(I suggest you take a closer look at the output during (re)installation)

Could you elaborate what is outputted when you (re)install plotly-resampler, as this compilation step takes place during installation?
I see that you are working on Windows; to compile C-code on Windows you need to have `gcc`, what happens when you type `gcc -v` in your powershell?

**Reply by MIKKELLORENZ** [reply]
**Ahh yes, that is a good hint.**

**When I install plotly-resampler with conda I'm using the Anaconda Prompt. There doesn't seem to be any mention of GCC:**

_`(base) C:\Users\new_user>conda install -c conda-forge plotly-resampler
Collecting package metadata (current_repodata.json): done
Solving environment: done

  environment location: C:\ProgramData\Anaconda3

  added / updated specs:
    - plotly-resampler


The following NEW packages will be INSTALLED:

  ansi2html          conda-forge/win-64::ansi2html-1.8.0-py39hcbf5309_1 None
  brotli-python      conda-forge/win-64::brotli-python-1.0.9-py39h415ef7b_7 None
  dash               conda-forge/noarch::dash-2.7.0-pyhd8ed1ab_0 None
  dash-core-compone~ conda-forge/noarch::dash-core-components-2.0.0-pyhd8ed1ab_1 None
  dash-html-compone~ conda-forge/noarch::dash-html-components-2.0.0-pyhd8ed1ab_1 None
  dash-table         conda-forge/noarch::dash-table-5.0.0-pyhd8ed1ab_1 None
  flask-compress     conda-forge/noarch::flask-compress-1.13-pyhd8ed1ab_0 None
  flask-cors         conda-forge/noarch::flask-cors-3.0.10-pyhd8ed1ab_0 None
  jupyter-dash       conda-forge/noarch::jupyter-dash-0.4.2-pyhd8ed1ab_1 None
  orjson             conda-forge/win-64::orjson-3.8.2-py39h6c6a30e_0 None
  plotly-resampler   conda-forge/noarch::plotly-resampler-0.8.1-pyhd8ed1ab_0 None
  retrying           conda-forge/noarch::retrying-1.3.3-py_2 None
  trace-updater      conda-forge/noarch::trace-updater-0.0.9-pyhd8ed1ab_0 None

The following packages will be UPDATED:

  conda              pkgs/main::conda-22.9.0-py39haa95532_0 --> conda-forge::conda-22.9.0-py39hcbf5309_2 None


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Retrieving notices: ...working... done`_

**When I do gcc -v in Powershell it returns:**

`PS C:\Users\new_user> gcc -v
gcc : The term 'gcc' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path
is correct and try again.
At line:1 char:1
+ gcc -v
+ ~~~
    + CategoryInfo          : ObjectNotFound: (gcc:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException`

**So it appear that is likely not installed. I wonder why conda doesn't suggest this. I guess first step is to find a  GNU installer first?**

**Reply by MIKKELLORENZ** [reply]
Wow I am bad at formatting text, please forgive xD

**Comment by jvdd** [comment]
Hi @MIKKELLORENZ 

Starting from version `0.9.0` we will use [tsdownsample](https://github.com/predict-idlab/tsdownsample) - which provides Python bindings for pre-compiled Rust code of highly optimized (i.e., extremely fast) downsampling functions.

You can already try out a pre-release here: https://pypi.org/project/plotly-resampler/0.9.0rc3/

Cheers, Jeroen
